src/_imagingft.c

```diff
@@ -1125,10 +1129,21 @@ font_getvaraxes(FontObject *self) {
     name_count = FT_Get_Sfnt_Name_Count(self->face);
 
     list_axes = PyList_New(num_axis);
+    if (list_axes == NULL) {
+        return NULL;
+    }
     for (i = 0; i < num_axis; i++) {
         axis = master->axis[i];
 
         list_axis = PyDict_New();
+        if (list_axis == NULL) {
+            for (j = 0; j < i; j++) {
+                list_axis = PyList_GetItem(list_axes, j);
+                Py_DECREF(list_axis);
+            }
+            Py_DECREF(list_axes);
```

Comment:
Feels like this duplicated cleaning code should be deduplicated, maybe..?

Reply:
Ok, I've pushed a commit to rearrange the code.

Reply:
@akx How's it look now?

Reply:
This loop looks suspicious to me. `PyList_GetItem` [returns a borrowed reference](https://docs.python.org/3/c-api/list.html#c.PyList_GetItem), so the decref will cause it to be freed. However, the list is expected to also free its contents when it's reference count is decrefed later: https://github.com/python/cpython/blob/e375bff03736f809fbc234010c087ef9d7e0d384/Objects/listobject.c#L356

So I would expect this loop to cause each `list_axis` to be double freed (except the last, which has not been added to the list yet).

Reply:
Thanks. I've applied the suggestion from @nulano, which means that the loop was de-duplicated is no longer present. So I've actually undone the rearranging of the code as it is no longer necessary.